plugins/cpp/service/src/filediagram.cpp

```diff
@@ -415,28 +415,47 @@ std::vector<util::Graph::Node> FileDiagram::getImplementedFiles(
   const util::Graph::Node& node_,
   bool reverse_)
 {
-  std::vector<util::Graph::Node> implements;
-
-  _transaction([&, this]{
-    EdgeResult res = _db->query<model::CppEdge>(
-      (reverse_
-         ? EdgeQuery::to->domainId
-         : EdgeQuery::from->domainId) == node_ &&
-      EdgeQuery::from->domainId != EdgeQuery::to->domainId &&
-      EdgeQuery::type == model::CppEdge::IMPLEMENT);
+  std::vector<core::FileId> implements;
+  _transaction([&, this]
+  {
+    auto contained = _db->query<model::File>(
+      odb::query<model::File>::parent == std::stoull(node_) &&
+      odb::query<model::File>::type != model::File::DIRECTORY_TYPE
+    );
+    std::unordered_set<core::FileId> used;
 
-    for (const model::CppEdge& edge : res)
+    for (const model::File &file : contained)
```

Comment:
const auto& file

Reply:
Why use auto when the exact type is recognizable?

Reply:
I agree with decreasing the usage of `auto`. I wrote about it in another pull request, let me quote it here:
> By the way I'm not a big fan of autos. I think we shouldn't use them unless the substituted type is extreme long. Their usage don't help as much as it makes the code shorter. It is easier to skip such an optimization opportunity and it is more time consuming to recover the type for comprehension purposes.